src/main/java/org/mifos/connector/channel/api/CollectionApiController.java

```diff
@@ -0,0 +1,31 @@
+package org.mifos.connector.channel.api;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.support.DefaultExchange;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RestController;
+import javax.servlet.http.HttpServletResponse;
+
+@RestController
+public class CollectionApiController implements CollectionApi {
+
+    @Autowired
+    private ProducerTemplate producerTemplate;
+
+    @Override
+    public String collection(String tenant, String requestBody, HttpServletResponse response) {
+
+        Exchange exchange = new DefaultExchange(producerTemplate.getCamelContext());
+        exchange.getIn().setBody(requestBody);
```

Comment:
move these repeated lines to a single function to make it more compact. Also, see whether we can use the Builder pattern to set details to the exchange object.